File: Difficulty: Easy/Segregate 0s and 1s/segregate-0s-and-1s.py
#User function Template for python3

class Solution:
    def segregate0and1(self, arr):
        # code here
        left = []
        right = []
        for i in range(len(arr)):
            if(arr[i] == 0):
                left.append(arr[i])
            else:
                right.append(arr[i])
        
        index = 0
        
        
        
        for i in range(len(left)):
            arr[index] = left[i]
            index+=1
            
        for i in range(len(right)):
            arr[index] = right[i]
            index+=1
            
        return arr
        

File: Difficulty: Easy/Segregate 0s and 1s/segregate-0s-and-1s.java
//{ Driver Code Starts


// Initial Template for Java

import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t;
        t = Integer.parseInt(br.readLine());
        while (t-- > 0) {
            String line = br.readLine();
            String[] tokens = line.split(" ");

            // Create an ArrayList to store the integers
            ArrayList<Integer> array = new ArrayList<>();

            // Parse the tokens into integers and add to the array
            for (String token : tokens) {
                array.add(Integer.parseInt(token));
            }

            int[] arr = new int[array.size()];
            int idx = 0;
            for (int i : array) arr[idx++] = i;

            new Solution().segregate0and1(arr);
            for (int i = 0; i < array.size(); i++) {
                System.out.print(arr[i] + " ");
            }
            System.out.println();
            // System.out.println("~");
        
System.out.println("~");
}
    }
}

// } Driver Code Ends


// User function Template for Java

class Solution {
    void segregate0and1(int[] arr) {
        // code here
        int count0 = 0;
        int count1 = 0;
        for(int i = 0; i< arr.length; i++){
            if(arr[i] == 0){
                count0++;
            }else{
                count1++;
            }
        }
        int index = 0;
        for(int i = 0; i< count0; i++){
            arr[index++] = 0;
        }
        for(int i = 0; i< count1; i++){
            arr[index++] = 1;
        }
    }
}
